hub/src/server/authentication.ts

```diff
@@ -126,18 +185,18 @@ export class V1Authentication {
     // associationToken and verifies that it authorizes the "outer"
     // JWT's address (`bearerAddress`)
 
-    let associationToken: TokenType
-    try {
-      associationToken = decodeToken(token)
-    } catch (e) {
-      throw new ValidationError('Failed to decode association token in JWT')
-    }
+    const payload = decodeTokenForPayload({
+      encodedToken: token, 
+      caller: 'checkAssociationToken', 
```

Comment:
This structure seems odd to me.  Why can't this code simply throw an error here, instead of having to pass some error context into another function?